platform/darwin/src/NSCoder+MGLAdditions.mm

```diff
@@ -0,0 +1,16 @@
+#import "NSCoder+MGLAdditions.h"
+
+#import "NSArray+MGLAdditions.h"
+
+@implementation NSCoder (MGLAdditions)
+
+- (void)mgl_encodeLocationCoordinates2D:(std::vector<CLLocationCoordinate2D>)coordinates forKey:(NSString *)key {
+    [self encodeObject:[NSArray mgl_coordinatesFromCoordinates:coordinates] forKey:key];
```

Comment:
I stand corrected. Apparently the [root cause](http://stackoverflow.com/a/12292033) is that CLLocationCoordinate2D is a typedef to an anonymous struct, so `@encode` can't do anything useful.

We could hard-code the type string `{dd}`, as many Swift developers have had to do, or we could define our own non-anonymous struct based on CLLocationCoordinate2D solely for the purpose of archiving, or we could eat the performance cost and use dictionaries here.